static/shared/js/typeahead.js

```diff
@@ -98,13 +98,26 @@ export function clean_query_lowercase(query) {
     return query;
 }
 
+export const is_unicode_emoji = (emoji) =>
+    emoji.reaction_type === "unicode_emoji" && Boolean(emoji.emoji_code);
+
+export const parse_unicode_emoji_code = (code) =>
+    code
+        .split("-")
+        .map((hex) => String.fromCodePoint(Number.parseInt(hex, 16)))
+        .join("");
```

Comment:
restoring this without the try-catch since it is now only used with appropriate checks